### ISSUE TITLE
typo leads to baffling error message

### ISSUE THREAD

**Issue by vpax** [zeek/zeek#1245]
The following:
```
event zeek_init()
    {
    print0.0000000000005;
    }
```
on try.zeek.org (per https://try.zeek.org/#/tryzeek/saved/461911) produces the error message:
```
warning in ././try.zeek, line 3: expression value ignored ({59ac:62ed:d40:8827:1ff5:3cd5:1718:8b9e})
```

If you run it directly, you at least get the additional information that makes it clear what's going on:
```
warning in <stdin>, line 1: no such host: print0.0000000000005
```
though it's still pretty surprising.  Perhaps this isn't worth fixing, but one natural thing would be to tighten up the logic around potential hostnames to require at least some letters in the purported TLD.

**Comment by jsiwek** [comment]
I found this thorough answer of what's technically vs. practically allowed that was interesting to read: https://stackoverflow.com/questions/9071279/number-in-the-top-level-domain

My takeaway: technically an all-numeric TLD is permitted by specs, but current practice would reject anyone applying for that.  If we changed our validity-checking to look for "at least one letter" that would be in sensible/practical agreement.  Although, in #1251, I decided to require "starting with a letter" since it's maybe a minor help in catching a mistake like `0.05f` meaning a "floating-point literal" in C++-land, but then being misinterpreted as a hostname in Zeek-land.
